src/main/java/io/kestra/plugin/googleworkspace/sheets/AbstractLoad.java

```diff
@@ -0,0 +1,156 @@
+package io.kestra.plugin.googleworkspace.sheets;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.kestra.core.models.annotations.PluginProperty;
+import io.kestra.core.runners.RunContext;
+import io.kestra.core.serializers.JacksonMapper;
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.constraints.NotNull;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.io.IOUtils;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.List;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+    title = "Read from a sheets"
+)
+public abstract class AbstractLoad extends AbstractSheet {
+
+    private final static ObjectMapper JSON_MAPPER = JacksonMapper.ofJson();
+    private final static ObjectMapper ION_MAPPER = JacksonMapper.ofIon();
+
+    @Schema(
+        title = "The spreadsheet unique id"
+    )
+    @NotNull
+    @PluginProperty(dynamic = true)
+    protected String spreadsheetId;
+
+    @Builder.Default
+    @Schema(
+        title = "Specifies if the first line should be the header (default: false)"
+    )
+    protected final Boolean header = false;
+
+    @Schema(
+        title = "Csv parsing options (Optional)."
+    )
+    @Builder.Default
+    private CsvOptions csvOptions = CsvOptions.builder().build();
+
+    @Schema(
+        title = "Schema to read avro objects (Optional).",
+        description = "If provided, task will read avro objects from this schema."
+    )
+    @PluginProperty(dynamic = true)
+    private String avroSchema;
+
+    @Schema(
+        title = "Format of input file.",
+        description = "If not provided task will programmatically try to find correct format, base on extension"
+    )
+    @PluginProperty(dynamic = true)
+    private Format format;
+
+    protected List<List<Object>> parse(RunContext runContext, String from) throws Exception {
```

Comment:
The file should always be from the internal storage, so it should've an URI (you can still use the extension to automatically found the format).

Moreover, you should use the output stream of the URI from the internal storage directly instead of creating a file locally.

Reply:
> Moreover, you should use the output stream of the URI from the internal storage directly instead of creating a file locally.

So i have to avoid any file creation?

Reply:
If possible yes, temporary files can eat up filesystem space pretty quickly